lib/plugins/keyGeneratorFactory.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+// TODO: should be instance! cannot use Nocca instance references now.
+
+var $q = require('q');
+var $extend = require('extend');
+
+module.exports = {
+    interface: 'keyGenerator',
+    id: 'keyGeneratorFactory',
+    name: 'Key Generator Factory',
+    constructor: KeyGeneratorFactory
+};
+
+function KeyGeneratorFactory (Nocca) {
+
+    var self = this;
+    
+    self.generateKey = generateKey;
+
+    self.incrementalKeyGenerator = incrementalKeyGenerator;
```

Comment:
Yes. It's basically what the keyGeneratorFactory did using the combine function. Now that we are using the plugin structure for this, we could do exactly what you're describing here automatically and allow people to construct keys from any combination of generators. 

The only thing that should be made very clear in the documentation is that order of the generators is maintained and no separation of collected data is enforced (possibly allowing plugins to modify earlier generated (parts of a) key(s).